pyccel/codegen/printing/ccode.py

```diff
@@ -445,6 +445,13 @@ def _print_LiteralImaginaryUnit(self, expr):
         self._additional_imports.add("complex")
         return '_Complex_I'
 
+    def _print_PythonLen(self, expr):
+        var = expr.arg
+        if var.rank > 0:
+            return self._print(var.shape[0])
+        else:
+            raise NotImplementedError("PythonLen not implemented for type {}".format(type(expr.arg)))
```

Comment:
Could it be useful to also print the expression `expr` in the error message?

Reply:
Fixed in 1f005cf